crates/net/network/src/transactions/mod.rs

```diff
@@ -891,6 +894,14 @@ where
 
         // Note: Assuming ~random~ order due to random state of the peers map hasher
         for (peer_idx, (peer_id, peer)) in self.peers.iter_mut().enumerate() {
+            match self.config.propogation_policy {
+                TransactionPropagationPolicy::All => {}
+                TransactionPropagationPolicy::Trusted => {
+                    if !matches!(peer.peer_kind, PeerKind::Trusted) {
+                        continue
+                    }
+                }
+            }
```

Comment:
I'd like to convert this into a function of `TransactionPropagationPolicy` this way we could extend this type with another variant that takes a Box dyn impl